arrow-libs/core/arrow-core-data/src/main/kotlin/arrow/core/MapK.kt

```diff
@@ -1,217 +1,219 @@
-      j: Map<Key, J>,
-      k: Map<Key, K>,
-      map: (Key, B, C, D, E, F, G, H, I, J, K) -> L
-    ): Map<Key, L> {
-      val destination = LinkedHashMap<Key, L>(b.size)
-      for ((key, bb) in b) {
-        Nullable.mapN(c[key], d[key], e[key], f[key], g[key], h[key], i[key], j[key], k[key]) { cc, dd, ee, ff, gg, hh, ii, jj, kk ->
-          map(key, bb, cc, dd, ee, ff, gg, hh, ii, jj, kk)
-        }?.let { l -> destination.put(key, l) }
-      }
-      return destination
+  inline fun <Key, B, C, D, E, F, G, H, I, J> mapN(
+    b: Map<Key, B>,
+    c: Map<Key, C>,
+    d: Map<Key, D>,
+    e: Map<Key, E>,
+    f: Map<Key, F>,
+    g: Map<Key, G>,
+    h: Map<Key, H>,
+    i: Map<Key, I>,
+    map: (Key, B, C, D, E, F, G, H, I) -> J
+  ): Map<Key, J> =
+    mapN(
+      b,
+      c,
+      d,
+      e,
+      f,
+      g,
+      h,
+      i,
+      emptyMap<Key, Unit>(),
+      emptyMap<Key, Unit>()
+    ) { key, bb, cc, dd, ee, ff, gg, hh, ii, _, _ ->
+      map(key, bb, cc, dd, ee, ff, gg, hh, ii)
     }
-  }
-}
 
-@Deprecated(MapKDeprecation, ReplaceWith("this"))
-fun <K, A> Map<K, A>.k(): MapK<K, A> = MapK(this)
+  inline fun <Key, B, C, D, E, F, G, H, I, J, K> mapN(
+    b: Map<Key, B>,
+    c: Map<Key, C>,
+    d: Map<Key, D>,
+    e: Map<Key, E>,
+    f: Map<Key, F>,
+    g: Map<Key, G>,
+    h: Map<Key, H>,
+    i: Map<Key, I>,
+    j: Map<Key, J>,
+    map: (Key, B, C, D, E, F, G, H, I, J) -> K
+  ): Map<Key, K> =
+    mapN(b, c, d, e, f, g, h, i, j, emptyMap<Key, Unit>()) { key, bb, cc, dd, ee, ff, gg, hh, ii, jj, _ ->
+      map(key, bb, cc, dd, ee, ff, gg, hh, ii, jj)
+    }
 
-@Deprecated("Deprecated, use nullable instead", ReplaceWith("Tuple2<K, A>>?.let { ... }"))
-fun <K, A> Option<Tuple2<K, A>>.k(): MapK<K, A> =
-  when (this) {
-    is Some -> mapOf(this.t).k()
-    is None -> emptyMap<K, A>().k()
+  inline fun <Key, B, C, D, E, F, G, H, I, J, K, L> mapN(
+    b: Map<Key, B>,
+    c: Map<Key, C>,
+    d: Map<Key, D>,
+    e: Map<Key, E>,
+    f: Map<Key, F>,
+    g: Map<Key, G>,
+    h: Map<Key, H>,
+    i: Map<Key, I>,
+    j: Map<Key, J>,
+    k: Map<Key, K>,
+    map: (Key, B, C, D, E, F, G, H, I, J, K) -> L
+  ): Map<Key, L> {
+    val destination = LinkedHashMap<Key, L>(b.size)
+    for ((key, bb) in b) {
+      Nullable.mapN(
+        c[key],
+        d[key],
+        e[key],
+        f[key],
+        g[key],
+        h[key],
+        i[key],
+        j[key],
+        k[key]
+      ) { cc, dd, ee, ff, gg, hh, ii, jj, kk ->
+        map(key, bb, cc, dd, ee, ff, gg, hh, ii, jj, kk)
+      }?.let { l -> destination.put(key, l) }
+    }
+    return destination
   }
-
-@Deprecated(MapKDeprecation, ReplaceWith("this.map { it.key to it.value }.toMap()"))
-fun <K, A> List<Map.Entry<K, A>>.k(): MapK<K, A> = this.map { it.key to it.value }.toMap().k()
-
-@Deprecated("Deprecated, use nullable instead", ReplaceWith("map[k]?.let { ... }"))
-fun <K, A> Map<K, A>.getOption(k: K): Option<A> = Option.fromNullable(this[k])
```

Comment:
I think this might have been prematurely deprecated, since Option is still available

Reply:
I was wondering that myself, but that would mean we should duplicate all APIs for `null` and `Option` which I also don't think is great.
Especially since the idea of `Option` is only to use it in certain use-cases, and you can always lift nullable to `Option` whenever you need to like you have to do with Kotlin's Std today.

Reply:
having that in the balance I think we could all live without this tiny extension 😂

Reply:
We do need to fix the incorrect `ReplaceWith` tho 🙈